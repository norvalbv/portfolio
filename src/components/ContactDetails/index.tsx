import React, { ReactElement } from 'react';
import { CONTACT_EMAIL, GITHUB, LINKED_IN } from 'constants/index';
import CV from 'assets/Benjamin Norval CV.pdf';

const contactDetails = [
  {
    id: 'email',
    key: 'Email',
    hrefLink: `mailto:${CONTACT_EMAIL}`,
    label: CONTACT_EMAIL,
  },
  {
    id: 'linkedin',
    key: 'Linked In',
    hrefLink: LINKED_IN,
    label: 'Benjamin Norval',
  },
  {
    id: 'github',
    key: 'GitHub',
    hrefLink: GITHUB,
    label: 'norvalbv',
  },
  {
    id: 'myCV',
    key: 'My CV',
    hrefLink: CV,
    label: 'Download Here',
  },
];

const ContactDetails = (): ReactElement => {
  return (
    <div className="flex flex-col gap-2 rounded-lg bg-white/70 p-6 dark:bg-dark-dark/30 lg:w-2/5">
      <h2 className="mb-6 underline underline-offset-8">Contact Details:</h2>
      <p className="mb-4">If you would like to contact me directly;</p>
      {contactDetails.map((contact) => (
        <span className="my-1 text-sm md:text-base" key={contact.id}>
          {contact.key}:&nbsp;
          <a
            rel="noreferrer"
            target="_blank"
            href={contact.hrefLink}
            className="my-4 underline transition duration-150 ease-linear hover:text-accent-secondary"
            download={contact.id === 'cv'}
          >
            {contact.label}
          </a>
        </span>
      ))}
      <p className="mt-6 text-sm">I look forward to hearing from you!</p>
    </div>
  );
};

export default ContactDetails;
