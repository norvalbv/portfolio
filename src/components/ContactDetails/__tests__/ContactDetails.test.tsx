import React from 'react';
import renderer from 'react-test-renderer';

import ContactDetails from '..';

describe('<ContactDetails />', () => {
  test('should render correctly ', () => {
    const tree = renderer.create(<ContactDetails />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
