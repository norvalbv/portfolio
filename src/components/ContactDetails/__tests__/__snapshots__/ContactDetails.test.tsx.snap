// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactDetails /> should render correctly  1`] = `
<div
  className="flex flex-col gap-2 rounded-lg bg-white/70 p-6 dark:bg-dark-dark/30 lg:w-2/5"
>
  <h2
    className="mb-6 underline underline-offset-8"
  >
    Contact Details:
  </h2>
  <p
    className="mb-6"
  >
    If you would like to contact me directly;
  </p>
  <span
    className="my-2"
  >
    Email
    : 
    <a
      className="my-4 underline transition duration-150 ease-linear hover:text-accent-secondary"
      download={false}
      href="mailto:benjinorval@gmail.com"
      rel="noreferrer"
      target="_blank"
    >
      benjinorval@gmail.com
    </a>
  </span>
  <span
    className="my-2"
  >
    Phone
    : 
    <a
      className="my-4 underline transition duration-150 ease-linear hover:text-accent-secondary"
      download={false}
      href="tel:+447508719043"
      rel="noreferrer"
      target="_blank"
    >
      (+44) 7508719043
    </a>
  </span>
  <span
    className="my-2"
  >
    Linked In
    : 
    <a
      className="my-4 underline transition duration-150 ease-linear hover:text-accent-secondary"
      download={false}
      href="https://www.linkedin.com/in/benjamin-norval/"
      rel="noreferrer"
      target="_blank"
    >
      Benjamin Norval
    </a>
  </span>
  <span
    className="my-2"
  >
    GitHub
    : 
    <a
      className="my-4 underline transition duration-150 ease-linear hover:text-accent-secondary"
      download={false}
      href="https://github.com/norvalbv/"
      rel="noreferrer"
      target="_blank"
    >
      norvalbv
    </a>
  </span>
  <span
    className="my-2"
  >
    CV
    : 
    <a
      className="my-4 underline transition duration-150 ease-linear hover:text-accent-secondary"
      download={true}
      href="test-file-stub"
      rel="noreferrer"
      target="_blank"
    >
      CV
    </a>
  </span>
  <p
    className="mt-6 italic"
  >
    I look forward to hearing from you!
  </p>
</div>
`;
